File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/MobileConfigurations/Welcome.cs
namespace Psi.Data.Models.ClientConfigurationModels.MobileConfigurations
{
	public class Welcome : SettingsBaseHelper
    {
	    public Welcome(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("Mobile.Welcome.MinimumVersion")]
        public string MinimumVersion
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Mobile.Welcome.MinimumAndroidVersion")]
        public string MinimumAndroidVersion
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Mobile.Welcome.MinimumIosVersion")]
        public string MinimumIosVersion
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Mobile.Welcome.Enabled")]
        public bool Enabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Mobile.Welcome.TokensEnabled")]
        public bool TokensEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Mobile.Welcome.Url")]
        public string Url
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }
	}
}

File: infrastructure/models/ClientConfigurationModels/MobileConfigurations/Welcome.ts
import { Setting, ISettingsGroup, ISettingsMetadata } from '@models/base/types';
export interface WelcomeConfig {
    MinimumVersion: string;
    MinimumAndroidVersion: string;
    MinimumIosVersion: string;
    Enabled: boolean;
    TokensEnabled: boolean;
    Url: string;
}

export class Welcome implements ISettingsGroup {
    static readonly metadata: ISettingsMetadata = {
        groupName: 'Welcome'
    };


            private _minimumVersion: string;
            get minimumVersion(): string {
                return this._minimumVersion;
            }
            set minimumVersion(value: string) {
                this._minimumVersion = value;
            }

            private _minimumAndroidVersion: string;
            get minimumAndroidVersion(): string {
                return this._minimumAndroidVersion;
            }
            set minimumAndroidVersion(value: string) {
                this._minimumAndroidVersion = value;
            }

            private _minimumIosVersion: string;
            get minimumIosVersion(): string {
                return this._minimumIosVersion;
            }
            set minimumIosVersion(value: string) {
                this._minimumIosVersion = value;
            }

            private _enabled: boolean;
            get enabled(): boolean {
                return this._enabled;
            }
            set enabled(value: boolean) {
                this._enabled = value;
            }

            private _tokensEnabled: boolean;
            get tokensEnabled(): boolean {
                return this._tokensEnabled;
            }
            set tokensEnabled(value: boolean) {
                this._tokensEnabled = value;
            }

            private _url: string;
            get url(): string {
                return this._url;
            }
            set url(value: string) {
                this._url = value;
            }


    constructor() {}


        toSettings(): Setting[] {
            return [
                { key: "Welcome.MinimumVersion", value: this._minimumVersion, dataType: 'string', label: "Minimum Version" },
                { key: "Welcome.MinimumAndroidVersion", value: this._minimumAndroidVersion, dataType: 'string', label: "Minimum Android Version" },
                { key: "Welcome.MinimumIosVersion", value: this._minimumIosVersion, dataType: 'string', label: "Minimum Ios Version" },
                { key: "Welcome.Enabled", value: this._enabled, dataType: 'boolean', label: "Enabled" },
                { key: "Welcome.TokensEnabled", value: this._tokensEnabled, dataType: 'boolean', label: "Tokens Enabled" },
                { key: "Welcome.Url", value: this._url, dataType: 'string', label: "Url" },
            ];
        }

}